pandas/core/indexes/category.py

```diff
@@ -169,7 +169,7 @@ def _create_categorical(cls, data, categories=None, ordered=None,
                 data = data.set_categories(categories, ordered=ordered)
             elif ordered is not None and ordered != data.ordered:
                 data = data.set_ordered(ordered)
-            if isinstance(dtype, CategoricalDtype):
+            if isinstance(dtype, CategoricalDtype) and dtype != data.dtype:
```

Comment:
It culd be noted that in master this comparison is quite slow:

```python
>>> d = pd.api.types.CategoricalDtype(categories=['a', 'b', 'c'])
>>> %timeit d == d
149 µs  # master
524 ns  # this PR
```

This is the reason for the focus on improving dtype comparisons, so this check doesn't cause slowdowns on other parts of pandas.